From 9428e56ae9f5429c186fd27dcb8a9bca202d26f9 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Sun, 30 May 2021 14:30:56 +0200
Subject: [PATCH] Replace werkzeuz's rename with os.replace; enable
 Werkzeug 2.x

This function has been always internal, and has been removed from
Werkzeug 2.x.

Python 3.3 provides os.replace, a cross-platform atomic rename function.

If this is unavailable, on POSIX we can fall back to posix.rename.

Ref: https://github.com/pallets/werkzeug/issues/1759
Ref: https://bugs.python.org/issue8828
Upstream-Status: Submitted [https://github.com/lektor/lektor/pull/911]
Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 lektor/builder.py | 5 ++++-
 lektor/utils.py   | 5 ++++-
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/lektor/builder.py b/lektor/builder.py
index 93572fd8f0dc..af9e6467ad16 100644
--- a/lektor/builder.py
+++ b/lektor/builder.py
@@ -11,7 +11,10 @@ from contextlib import contextmanager
 from itertools import chain
 
 import click
-from werkzeug.posixemulation import rename
+try:
+    from os import replace as rename
+except ImportError:
+    from posix import rename
 
 from lektor.build_programs import builtin_build_programs
 from lektor.buildfailures import FailureController
diff --git a/lektor/utils.py b/lektor/utils.py
index 2402a0aded72..7565b8cc7354 100644
--- a/lektor/utils.py
+++ b/lektor/utils.py
@@ -25,8 +25,11 @@ from markupsafe import Markup
 from slugify import slugify as _slugify
 from werkzeug import urls
 from werkzeug.http import http_date
-from werkzeug.posixemulation import rename
 from werkzeug.urls import url_parse
+try:
+    from os import replace as rename
+except ImportError:
+    from posix import rename
 
 
 is_windows = os.name == "nt"
-- 
2.31.1

